test/button.element.test.ts

```diff
@@ -52,4 +53,133 @@ describe('sgds-button', () => {
     expect(el.shadowRoot?.querySelector("button")).not.to.exist;
   })
 
+  it('should emit a click event when calling click()', async () => {
+    const el = await fixture<SgdsButton>(html` <sgds-button></sgds-button> `);
+    const clickHandler = sinon.spy();
+
+    el.addEventListener('click', clickHandler);
+    el.click();
+    await waitUntil(() => clickHandler.calledOnce);
+
+    expect(clickHandler).to.have.been.calledOnce;
+  });
+
+});
+
+
+
+describe('when submitting a form', () => {
+  it('should submit when the button is inside the form', async () => {
+    const form = await fixture<HTMLFormElement>(html`
+      <form action="" method="post">
+        <sgds-button type="submit"></sgds-button>
+      </form>
+    `);
+    const button = form.querySelector<SgdsButton>('sgds-button')!;
+    const handleSubmit = sinon.spy((event: SubmitEvent) => event.preventDefault());
+
+    form.addEventListener('submit', handleSubmit);
+    button.click();
+    expect(handleSubmit).to.have.been.calledOnce;
+  });
+
+  it('should submit when the button is outside the form and has a form attribute', async () => {
+    const el = await fixture(html`
+      <div>
+        <form id="a" action="" method="post"></form>
+        <sgds-button type="submit" form="a">Submit</sgds-button>
+      </div>
+    `);
+    const form = el.querySelector<HTMLFormElement>('form')!;
+    const button = el.querySelector<SgdsButton>('sgds-button')!;
+    const handleSubmit = sinon.spy((event: SubmitEvent) => event.preventDefault());
+
+    form.addEventListener('submit', handleSubmit);
+    button.click();
+
+    expect(handleSubmit).to.have.been.calledOnce;
+  });
+
+  it('should override form attributes when formaction, formmethod, formnovalidate, and formtarget are used inside a form', async () => {
```

Comment:
what will happen if its this case ?
```
   <form id="a" action="foo" method="post" target="_self">
        <sgds-button type="submit" form="a">
          Submit
        </sgds-button>
      </form>
```
should one expect the following?
```
   expect(handleSubmit).to.have.been.calledOnce;
    expect(submitter.formAction.endsWith('/foo')).to.be.true;
    expect(submitter.formMethod).to.equal('post');
    expect(submitter.formTarget).to.equal('_self');
    expect(submitter.formNoValidate).to.be.true;
```

Reply:
I `expect(submitter.formNoValidate).to.be.true; `
to fail since formnovalidate attribute is not declared in sgds-button or form.